en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-websocket-api-in-choreo-connect.md

```diff
@@ -1,43 +1,108 @@
+!!! info "Before you begin"
+
+    This guide assumes that you already have a Choreo Connect instance configured to run with API Manager. If not, checkout the [Quick Start Guide]({{base_path}}/deploy-and-publish/deploy-on-gateway/choreo-connect/getting-started/quick-start-guide-docker-with-apim/) on how to install and run Choreo Connect. To learn more about Choreo Connect, have a look at the [Overview of Choreo Connect]({{base_path}}/deploy-and-publish/deploy-on-gateway/choreo-connect/getting-started/choreo-connect-overview).
 
 
 ### Step 1 - Create a WebSocket API in API Manager
 
- For instructions on how to create a WebSocket API, see [Create a WebSocket API]({{base_path}}/design/create-api/create-streaming-api/create-a-websocket-streaming-api/).
+1. Create a WebSocket API by following the steps in [Create a WebSocket API]({{base_path}}/design/create-api/create-streaming-api/create-a-websocket-streaming-api/).
+
+2. By default, a WebSocket API is protected by OAuth2 in Choreo Connect. Make sure to switch on security for the API via Publisher and **Save**.
 
-### Step 2 - Deploy the API in the Choreo Connect environment
+    [![Security switched on]({{base_path}}/assets/img/design/create-api/streaming-api/streaming-api-security-on.png)]({{base_path}}/assets/img/design/create-api/streaming-api/streaming-api-security-on.png)
 
-For more information on deploying the API in Choreo Connect, see [Deploy API]({{base_path}}/deploy-and-publish/deploy-on-gateway/deploy-api/deploy-an-api/).
+### Step 2 - Deploy and publish the API
+
+1. Deploy the API in Choreo Connect by navigating to the **Deployments** page from the left menu. For more information, see [Deploy API]({{base_path}}/deploy-and-publish/deploy-on-gateway/deploy-api/deploy-an-api/).
+2. Publish the API from the **Lifecycle** page.
 
 ### Step 3 - Generate an Access Token to invoke the API
 
-By default, the WebSocket API is protected by an OAuth2 token.
+!!! tip 
+    To generate a temporary test key to invoke the API follow the steps [here]({{base_path}}/deploy-and-publish/deploy-on-gateway/choreo-connect/security/generate-a-test-jwt/)
+
+1. Click **View in Dev Portal** at the top right corner to open Developer Portal in another browser tab.
+
+2. Open **Applications** from the top menu and select **DefaultApplication** from the list.
+
+3. Open the **Subscriptions** tab and subscribe your API to the **DefaultApplication**.
 
-For more information on generating a JWT Access token, see [Get a Test Key to Invoke an API]({{base_path}}/consume/invoke-apis/invoke-apis-using-tools/invoke-an-api-using-the-integrated-api-console/#get-a-test-key-to-invoke-an-api).
+4. Open **APIs** from the top menu and select your API.
+
+5. Open the **Subscriptions** tab from the left menu bar, click on **PROD KEYS**, and generate keys.
+
+6. Open the **Try Out** tab and click **GET TEST KEY**. This will include the access token in the cURL command you generate in the section below
```

Comment:
```suggestion
6. Open the **Try Out** tab and click **GET TEST KEY**. This will include the access token in the cURL command you generate in the section below.
```